### PR TITLE
Max length check/handling for SyncTriggers

### PR DESCRIPTION
**Pull request by mathewc**
The Front End enforces a max sync triggers payload size. When exceeded it returns 400 BadRequest. We should perform this check ourselves before sending the request, and if we're over size, don't send the extended format.

### CODE REVIEW

File: src/WebJobs.Script/Utility.cs [hunk 1]
```diff
@@ -278,8 +278,12 @@ public static bool IsJson(string input)
...
 
             input = input.Trim();
-            return (input.StartsWith("{", StringComparison.OrdinalIgnoreCase) && input.EndsWith("}", StringComparison.OrdinalIgnoreCase))
-                || (input.StartsWith("[", StringComparison.OrdinalIgnoreCase) && input.EndsWith("]", StringComparison.OrdinalIgnoreCase));
+            return IsJsonObject(input) || (input.StartsWith("[", StringComparison.OrdinalIgnoreCase) && input.EndsWith("]", StringComparison.OrdinalIgnoreCase));
+        }
+
+        public static bool IsJsonObject(string input)
+        {
+            return input.Trim().StartsWith("{", StringComparison.OrdinalIgnoreCase) && input.EndsWith("}", StringComparison.OrdinalIgnoreCase);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs]: is this enough to check for a Json Object?  Check if parsing the string succeeds: JToken.Parse(input)?

File: src/WebJobs.Script/Utility.cs [hunk 2]
```diff
@@ -278,8 +278,12 @@ public static bool IsJson(string input)
             }
 
             input = input.Trim();
-            return (input.StartsWith("{", StringComparison.OrdinalIgnoreCase) && input.EndsWith("}", StringComparison.OrdinalIgnoreCase))
-                || (input.StartsWith("[", StringComparison.OrdinalIgnoreCase) && input.EndsWith("]", StringComparison.OrdinalIgnoreCase));
+            return IsJsonObject(input) || (input.StartsWith("[", StringComparison.OrdinalIgnoreCase) && input.EndsWith("]", StringComparison.OrdinalIgnoreCase));
+        }
+
+        public static bool IsJsonObject(string input)
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs]: Add unit tests for this method

File: src/WebJobs.Script/Utility.cs [hunk 3]
```diff
@@ -278,8 +278,12 @@ public static bool IsJson(string input)
             }
 
             input = input.Trim();
-            return (input.StartsWith("{", StringComparison.OrdinalIgnoreCase) && input.EndsWith("}", StringComparison.OrdinalIgnoreCase))
-                || (input.StartsWith("[", StringComparison.OrdinalIgnoreCase) && input.EndsWith("]", StringComparison.OrdinalIgnoreCase));
+            return IsJsonObject(input) || (input.StartsWith("[", StringComparison.OrdinalIgnoreCase) && input.EndsWith("]", StringComparison.OrdinalIgnoreCase));
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs]: Move checking for JArray to the helper as well.

File: src/WebJobs.Script/Utility.cs (hunk 1 above)
Comment (mathewc) [reply src/WebJobs.Script/Utility.cs]: Used JToken.Parse - good idea :)

File: src/WebJobs.Script/Utility.cs (hunk 2 above)
Comment (mathewc) [reply src/WebJobs.Script/Utility.cs]: Reverted these changes

**Pull request merged**